src/instrument.ts

```diff
@@ -0,0 +1,262 @@
+  const origOpts = opts;
+  opts = opts || {};
+  if (typeof origOpts === 'string') {
+    opts.sql = origOpts as string;
+  }
+
+  return getTracer(opts).startActiveSpan(
+    SPAN_PREFIX + '/' + spanNameSuffix,
+    {kind: SpanKind.CLIENT},
+    span => {
+      span.setAttribute(SEMATTRS_DB_SYSTEM, 'spanner');
+
+      if (opts.tableName) {
+        span.setAttribute(SEMATTRS_DB_SQL_TABLE, opts.tableName);
+      }
+      if (opts.dbName) {
+        span.setAttribute(SEMATTRS_DB_NAME, opts.dbName);
+      }
+
+      const definedExtendedTracing = opts.enableExtendedTracing !== undefined;
+      // If they optedInPII but opts.enableExtendedTracing=false, reject it.
+      const explicitlySkipET =
+        definedExtendedTracing && !opts.enableExtendedTracing;
+      if (
+        opts.sql &&
+        !explicitlySkipET &&
+        (opts.enableExtendedTracing || optedInPII)
+      ) {
+        const sql = opts.sql;
+        if (typeof sql === 'string') {
+          span.setAttribute(SEMATTRS_DB_STATEMENT, sql as string);
+        } else {
+          const stmt = sql as SQLStatement;
+          span.setAttribute(SEMATTRS_DB_STATEMENT, stmt.sql);
+        }
+      }
+
+      if (opts.that) {
+        const fn = cb.bind(opts.that);
+        return fn(span);
+      } else {
+        return cb(span);
+      }
+    }
+  );
+}
+
+// setSpanError sets err, if non-nil onto the span with
+// status.code=ERROR and the message of err.toString()
+export function setSpanError(span: Span, err: Error | String): boolean {
+  if (!err || !span) {
+    return false;
+  }
+
+  span.setStatus({
+    code: SpanStatusCode.ERROR,
+    message: err.toString(),
+  });
+  return true;
+}
+
+export function setSpanErrorAndException(span: Span, err: Error | String) {
+  if (setSpanError(span, err)) {
+    span.recordException(err as Error);
+  }
+}
+
+export function setGlobalContextManager(manager: ContextManager) {
+  context.setGlobalContextManager(manager);
+}
+
+export function disableContextAndManager(manager: ContextManager) {
+  manager.disable();
+  context.disable();
+}
+
+// getActiveOrNoopSpan queries the tracer for the currently active span
+// and returns it, otherwise if there is no active span available, it'll
+// simply create a NoopSpan. This is important in the cases where we don't
+// want to create a new span such is sensitive and frequently called code
+// for which the new spans would be too many and thus pollute the trace,
+// but yet we'd like to record an important annotation.
+export function getActiveOrNoopSpan(): Span {
+  const span = trace.getActiveSpan();
+  if (span) {
+    return span;
+  }
+  return new noopSpan();
+}
+
+// noopSpan is a pass-through Span that does nothing and shall not
+// be exported, nor added into any context. It serves as a placeholder
+// to allow calls in sensitive areas like sessionPools to transparently
+// add attributes to spans without lots of ugly null checks.
+class noopSpan implements Span {
```

Comment:
At least Java opentelmetry already has a Noop span implementation , dont we have something similar for Node ?

https://open-telemetry.github.io/opentelemetry-js/classes/_opentelemetry_sdk_trace_base.NoopSpanProcessor.html

Reply:
I finally found it under "NonRecordingSpan"

Reply:
What does this mean ? Can we use `NonRecordingSpan` directly instead of creating our `noopSpan` class ?

Reply:
Sorry I hadn't completed my response. So NonRecordingSpan isn't exported out of use of opentelemetry as a visible symbol sadly.

Reply:
Can you please share the link of NonRecordingSpan code, I just want to make sure we have used the same code for the same.